File: producer/update.py
from datetime import datetime


class Update:
    content: str
    author_id: str
    creation_date: datetime
    url: str
    media: list

    def __init__(
            self,
            content: str,
            author_id: str,
            creation_date: datetime,
            url: str,
            media: list):
        self.content = content
        self.author_id = author_id
        self.creation_date = creation_date
        self.url = url
        self.media = media

File: Common/Update.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Common
{
    public record Update
    {
        public string Content { get; init; }

        public string AuthorId { get; init; }

        public DateTime? CreationDate { get; init; }

        public string Url { get; init; }

        public List<IMedia> Media { get; init; }

        public bool Repost { get; init; }
        
        public string Source { get; init; }

        public override string ToString()
        {
            return $"{Url} (By {AuthorId}, Media length: {Media?.Count}, Repost: {Repost})";
        }
    }
}